src/libsystemd/sd-journal/journal-file.c

```diff
@@ -2917,11 +2918,13 @@ static int generic_array_bisect_step(
                 if (i == 0)
                         return TEST_GOTO_PREVIOUS;
 
-                /* Otherwise, cutting the array short. So, here we limit the number of elements we will see
-                 * in this array, and set the right boundary to the last possibly non-corrupted object. */
+                /* Otherwise, cutting the array short, and restart the bisection. So, here we limit the
+                 * number of elements we will see in this array, and set the right boundary to the last
+                 * possibly non-corrupted object. */
                 *m = i;
+                *left = 0;
```

Comment:
Hmm, but do we need to set this all the way to 0? If we already checked earlier that `*left` is at some value, we could just set `*left` to `MIN(*left, *right)`?